### PR TITLE
Objects instead of json

### PR DESCRIPTION
**Pull request by MakeShiftArtist**
# New Objects
Still a work in progress but I didn't like being forced to look at json data to see what was happening, so I added some objects to do that work for us. I did some bug testing and everything seems to be running smoothly. The objects should be fairly well documented as well
## User
Meant for user objects, not artists
```py
client.me()
```
Now returns a User object instead of some json data
## Artist
The artist object. Given that some info varies depending on how the request is made, i bug tested and it SHOULD work fine, but you may have some `None` values
## Album
The album object. there may be some None values, but there are 4 new methods to get additional info as well.
## Track
the track object, again there may be some None values but it has 3 new methods to get more infomation.

# New search methods
if a dev wants to search for artists, tracks, or albums, they can now use `client.search_artists`, `client.search_track`, and `client.search_albums`, respectively, which are generators that give the appropriate objects

# Example code
```py
import spotipy
import json
from spotipy.oauth2 import SpotifyOAuth

with open('config.json') as f:
  config = json.load(f)

client = spotipy.Spotify(
    auth_manager=SpotifyOAuth(
        client_id=config["client_id"],
        client_secret=config["client_secret"],
        redirect_uri="http://localhost/",
        scope="user-library-read"
    )
)

albums = client.search_albums("Perception", limit=1)
for album in albums:
    for track in album.get_tracks():
        print(track)
        print(track.explicit)
        break

"""
Intro III
False
"""
``

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 setup(
     name='spotipy',
-    version='2.19.0',
+    version='2.19.1',
```
Comment (stephanebruckert) [inline setup.py:21]: versions are bumped independently from the PRs

**Review by stephanebruckert — COMMENTED** [review]
Hey! Sorry about the late review. I appreciate the work you put in in this PR but I see several issues with it:

 - structuring responses is nice, but to me it goes against the "light-weight" vision of spotipy where, we are supposedly confident to deal with raw responses in JSON. The benefit being that if the Spotify API publishes a new field, it will be available "as usual" and without having to modify the library. With the changes in this PR, I'm not sure we would benefit from that. I'm happy to be wrong about this and change my mind, but this is how I see it now
 - this looks backward-incompatible to me. If we are going to do this let's discuss it in the version 3 discussion https://github.com/plamere/spotipy/issues/652
 - the change only affects the search functions, while I believe if we were to do this, we should do it for all methods at once, everywhere

**Comment by IdmFoundInHim** [comment]
The object-based structure is used by [Tekore](https://github.com/felix-hilden/tekore), another Spotify API wrapper under active development. I would recommend looking there, as this repo is committed to keeping the JSON output. (There was a [fairly](https://github.com/plamere/spotipy/issues/387) [substantial](https://github.com/plamere/spotipy/issues/404) [discussion](https://github.com/plamere/spotipy/issues/406) that led to the split of Tekore and Spotipy development to provide two different approaches.)

**Comment by stephanebruckert** [comment]
Closing this as wont-fix.

Other than spotipy being light-weight and python being a straight-forward and to-the-point language, my main argument is that objectifying API responses can lead to making users believe a field isn't available when it is actually part of the returned response. Returning JSON will always tell "the truth".

**Pull request closed without merging**